### DISCUSSION TITLE
Get password after OAuth sign in

### DISCUSSION THREAD

**Discussion by davidbrochart** [fief-dev/fief#232] [Q&A] [answered]
Say I successfully authenticate using GitHub OAuth, and this creates a user in Fief with the user's email provided by GitHub. If I open a new "incognito" window (without any cookie), I will need to go through the same flow again: sign in to GitHub and finalize sign up. But if I enter the same email address again, Fief says that "a user with the same email address already exists", which is true.
The second time, I could have signed in providing my email, but since I never received a password the first time, I need to click on "I forgot my password" in order to enter one. This works, but I'm wondering if it could be easier. For instance, the first time I sign in with GitHub, maybe there could be a option to see or send the password when finalizing the sign up?

**Comment by frankie567** [comment, marked as answer]
* Normally, if you sign up through GitHub a first time, a user and an associated OAuth Account is created in the DB. If you try to sign in/sign up again with GitHub (even without cookies), you should directly be authenticated; Fief shouldn't ask you to register again. If that's not the case, then something is going wrong 🙃 
* Regarding sign in with the email if you used a social provider the first time, yes, that's normal. We generate a random password, but for security reasons, we can't share it directly with the user. In future versions, I plan to make email/password an optional authentication method that can be disabled, so users won't necessarily have a password.

**Reply by davidbrochart** [reply]
> * If you try to sign in/sign up again with GitHub (even without cookies), you should directly be authenticated; Fief shouldn't ask you to register again.

Yes, I can see that now. Not sure why I didn't have the same behavior previously.
Anyway, that doesn't seem to be an issue anymore, thanks!
